[GStreamer][Realtek] Enable 'lowdelay-mode' for realtek omx decoder

vivienne-w opened the pull request:
#### 63f8f7abbc23b8e3b0b984bd63991fe7659a5a48
<pre>
[GStreamer][Realtek] Enable &apos;lowdelay-mode&apos; for realtek omx decoder
<a href="https://bugs.webkit.org/show_bug.cgi?id=257121">https://bugs.webkit.org/show_bug.cgi?id=257121</a>

Reviewed by Philippe Normand.

Enabling low delay mode will force the video decoder to decode frames
one by one without holding a reference frame.

Original author: Manoj Bhatta &lt;manoj_bhatta@comcast.com&gt;
See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1087">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1087</a>

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::configureElementPlatformQuirks): Enable &apos;lowdelay-mode&apos; property

Canonical link: <a href="https://commits.webkit.org/264325@main">https://commits.webkit.org/264325@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67b5cd02f81db43b20aec4da804ff745d928724e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7416 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7678 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7850 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9049 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7621 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7427 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/9597 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7600 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10507 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7547 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8654 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6869 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9158 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5964 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6747 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6861 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10154 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7353 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/6011 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6701 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1745 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10910 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7088 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -2296,9 +2296,15 @@ void MediaPlayerPrivateGStreamer::configureElementPlatformQuirks(GstElement* ele
-        GST_INFO_OBJECT(pipeline(), "Enable 'immediate-output' in rtkaudiosink");
-        g_object_set(element, "media-tunnel", FALSE, "audio-service", TRUE, "lowdelay-sync-mode", TRUE, nullptr);
+    if (m_streamPrivate) {
+        if (g_object_class_find_property(G_OBJECT_GET_CLASS(element), "media-tunnel")) {
```
`gstObjectHasProperty(element, "media-tunnel")`

vivienne-w committed:
[GStreamer][Realtek] Enable 'lowdelay-mode' for realtek omx decoder
https://bugs.webkit.org/show_bug.cgi?id=257121

Reviewed by Philippe Normand.

Enabling low delay mode will force the video decoder to decode frames
one by one without holding a reference frame.

Original author: Manoj Bhatta <manoj_bhatta@comcast.com>
See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1087

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::configureElementPlatformQuirks): Enable 'lowdelay-mode' property

Canonical link: https://commits.webkit.org/264325@main
